subprojects/gradle/signer/src/main/kotlin/com/avito/plugin/SignViaServiceAction.kt

```diff
@@ -74,50 +54,29 @@ internal class SignViaServiceAction(
         return Request.Builder()
             .url(hostWithoutSlash + apiPath)
             .post(body)
-            .tag(RequestMetadata::class.java, RequestMetadata("signer", "sign"))
+            .tag(
+                RequestMetadata::class.java,
+                RequestMetadata("signer", "sign")
+            )
             .build()
     }
 
-    private fun Response.process(outputFile: File) {
-        if (this.isSuccessful) {
+    private fun writeResponse(response: Response, outputFile: File) {
+        if (response.isSuccessful) {
             outputFile.createOrClear()
-            this.body
+            response.body
                 ?.byteStream()
                 ?.use { input ->
                     outputFile.outputStream().use { output ->
                         input.copyTo(output)
                     }
                 }
         } else {
-            val stringBody = this.body
+            val stringBody = response.body
                 ?.string()
                 ?: "Cannot read the response body"
 
-            error("Sign service returned: ${this.code}, body: $stringBody")
+            error("Failed to sign APK via service: code ${response.code}, body: $stringBody")
         }
     }
-
-    private fun <T> retry(
```

Comment:
Reuse an interceptor instead of a custom retry

Reply:
thanks!, don't know why we needed that in first place,
as far as i remember we had `RetryInterceptor` already when added that retry